### PR TITLE
feat: Support multiple types of timestamps in the Time Component

### PR DESCRIPTION
**Pull request by kof**
## Description

Date can be in various forms, lets support them all:

1698377386 - A valid 10-digit Unix timestamp (October 26, 2023).
2147483647 - The maximum 32-bit signed integer timestamp (January 19, 2038).
1698377386000 - A valid 13-digit Unix timestamp (October 26, 2023).
9999999999999 - A far future date (November 20, 2286).
2024.10

## Steps for reproduction

1. click button
2. expect xyz

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env` file

### CODE REVIEW

**Review by johnsicili — APPROVED** [review]
Works. Thanks for the assist!

File: packages/sdk-components-react/src/time.test.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  expect(parseDate("1724938577")).toEqual(new Date("2024-08-29T13:36:17.000Z"));
+});
+
+test("maximum 32-bit signed integer timestamp", () => {
+  expect(parseDate("2147483647")).toEqual(new Date("2038-01-19T03:14:07.000Z"));
+});
+
+test("far future date", () => {
+  expect(parseDate("9999999999999")).toEqual(
+    new Date("2286-11-20T17:46:39.999Z")
```
Comment (TrySound) [inline packages/sdk-components-react/src/time.test.ts:20]: Huh, crazy year

File: packages/sdk-components-react/src/time.tsx [hunk 2]
```diff
@@ -327,6 +327,23 @@ const timeStyleOrUndefined = (
...
 };
 
+export const parseDate = (datetimeString: string) => {
+  if (datetimeString === "") {
+    return;
+  }
+  const isNumber = /^\d+$/.test(datetimeString);
+  let timestamp = Number(datetimeString);
+  if (isNumber && datetimeString.length === 10) {
+    timestamp = timestamp * 1000;
```
Comment (TrySound) [inline packages/sdk-components-react/src/time.tsx:337]: *=

File: packages/sdk-components-react/src/time.tsx [hunk 3]
```diff
@@ -327,6 +327,23 @@ const timeStyleOrUndefined = (
   return undefined;
 };
 
+export const parseDate = (datetimeString: string) => {
+  if (datetimeString === "") {
+    return;
+  }
+  const isNumber = /^\d+$/.test(datetimeString);
```
Comment (istarkov) [inline packages/sdk-components-react/src/time.tsx:334]: 01 is bad number
moreover it's check on positive Int can be done using JS i.e 
ordinary check isNumber +   Number.isInteger(value) && value >= 0;

File: packages/sdk-components-react/src/time.tsx [hunk 4]
```diff
@@ -327,6 +327,23 @@ const timeStyleOrUndefined = (
...
+    timestamp = timestamp * 1000;
+  }
+  const date = new Date(isNumber ? timestamp : datetimeString);
+  const isValid = false === Number.isNaN(date.getTime());
+
+  if (isValid) {
+    return date;
+  }
+};
+
```
Comment (istarkov) [inline packages/sdk-components-react/src/time.tsx:346]: IMO better to check that Date isValid first and only then try any other formats.
Or explain why we check non std date formats first

**Review by istarkov — APPROVED** [review]
anyway with tests it's ok

File: packages/sdk-components-react/src/time.tsx (hunk 4 above)
Comment (istarkov) [reply packages/sdk-components-react/src/time.tsx:346]: Why as Date with 10 digits string is NaN

Comment (istarkov) [reply packages/sdk-components-react/src/time.tsx:346]: `new Date("1000000000")` <- INVALID

**Pull request merged**